src/main/java/calculator/Calculator.java

```diff
@@ -0,0 +1,65 @@
+package calculator;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Calculator {
+
+    public static String REGEX = ",|:";
+
+    public static int splitAndSum(String text) {
+        if (isBlank(text)) {
+            return 0;
+        }
+        text = getTextWithOutCustomGroup(text);
+        return sum(toInts(split(text)));
+    }
+
+    private static String getTextWithOutCustomGroup(String text) {
+        Pattern r = Pattern.compile("//(.)\n(.*)");
+        Matcher m = r.matcher(text);
+        if (m.find()) {
+            addRegex(m);
+            return m.group(2);
+        }
+        return text;
+    }
+
+    private static void addRegex(Matcher m) {
+        REGEX += "|" + m.group(1);
+    }
+
+    private static String[] split(String text) {
+        return text.split(REGEX);
+    }
+
+    private static boolean isBlank(String text) {
+        return text == null || text.isBlank();
+    }
+
+    private static int sum(int[] numbers) {
+        int sum = 0;
+        for (int number : numbers) {
+            sum += number;
+        }
+        return sum;
+    }
+
+    private static int[] toInts(String[] values) {
+        int[] numbers = new int[values.length];
+        for (int i = 0; i < values.length; i++) {
+            checkDegitOrMinus(values[i]);
+            numbers[i] = Integer.parseInt(values[i]);
+        }
+        return numbers;
+    }
+
+    private static void checkDegitOrMinus(String value) {
+        if (!value.matches("-?\\d+")) {
+            throw new RuntimeException("숫자 타입이 아닙니다.");
+        }
+        if (Integer.parseInt(value) < 0) {
+            throw new RuntimeException("음수입니다.");
+        }
```

Comment:
```suggestion
        if (!value.matches("-?\\d+")) {
            throw new RuntimeException("숫자 타입이 아닙니다.");
        }
        if (Integer.parseInt(value) < 0) {
            throw new RuntimeException("음수입니다.");
        }
```
해당 책임을 가지는 `PositiveNumber` 객체가 있다면 좋을것 같습니다!

Reply:
어떠한 말씀이신지 잘 이해가 가지 않습니다..ㅠ 혹시 간단한 예시를 주실 수 있으실까요??😢

Reply:
저희만의 양수타입을 만드는것이죠! ㅎㅎ

이런 느낌으로요!
추가적으로 궁금한점은 DM주세요~
```
class PositiveNumber {
     생성자(String number) {
          // 검증하기 (숫자가 아니면 에러 발생 or 음수이면 에러발생
          if (Integer.parseInt(number) < 0 || ...) {
            throw new RuntimeException("양수만 가능합니다.");
        }
          this.number = number;
     }
}
```